backend/entityservice/database/insertions.py

```diff
@@ -89,23 +90,33 @@ def insert_encodings_into_blocks(db, dp_id: int, block_ids: List[List[str]], enc
         will require more local memory, but could be faster due to less network transfers.
 
     """
-    encodings_insertion_query = "INSERT INTO encodings (dp, encoding_id, encoding) VALUES %s"
-    blocks_insertion_query = "INSERT INTO encodingblocks (dp, encoding_id, block_id) VALUES %s"
-    encoding_data = ((dp_id, eid, encoding) for eid, encoding in zip(encoding_ids, encodings))
-
-    def block_data_generator(encoding_ids, block_ids):
-        for eid, block_ids in zip(encoding_ids, block_ids):
+    ## first we have to look up the block numbers corresponding to the given block_names
+    block_info_iter = get_block_metadata(db, dp_id)
+    block_lookup = {bl_name: bl_id for bl_name, bl_id, _ in block_info_iter}
+
+    encodings_insertion_query = "INSERT INTO encodings (encoding_id, encoding) VALUES %s"
+    blocks_insertion_query = "INSERT INTO encodingblocks (dp, entity_id, encoding_id, block_id) VALUES %s"
+    # we differentiate between entity_id and encoding_id.
+    # The entity_id is the id that the dataprovider assigns to an entity. Usually the row number of that entity in the
+    # dataset.
+    # The encoding_id is used internally to address encodings uniquely.
```

Comment:
Is the intention that there will be multiple `encoding-ids` per `entity-id`? Or is the `encoding_id` just a primary key for the encodings table?

I suggest renaming one or both to make it clearer.  Perhaps `encoding_pk` instead of `encoding_id`.